crm_l10n_eu_nace/models/crm_lead.py

```diff
@@ -0,0 +1,24 @@
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    nace_id = fields.Many2one(
+        comodel_name="res.partner.nace", string="Main NACE", index=True
+    )
+    secondary_nace_ids = fields.Many2many(
+        comodel_name="res.partner.nace", string="Secondary NACE"
+    )
+
+    @api.multi
+    def _create_lead_partner_data(self, *args, **kwargs):
+        """Propagate NACE activity to created partner."""
+        result = super()._create_lead_partner_data(*args, **kwargs)
+        if self.nace_id:
+            result["nace_id"] = self.nace_id.ids
+        if self.secondary_nace_ids:
+            result["secondary_nace_ids"] = [
+                (4, self.secondary_nace_ids.ids, 0)
```

Comment:
```suggestion
                (4, x) for x in self.secondary_nace_ids.ids
```

Reply:
Don't you always need the 3rd element in the tuple?

Reply:
Not really, but that's not the main problem here: you have to add a `(4, ...` command for each element.

Reply:
This is still not changed